From 735e49e62dc07cbe94cb75e9069a01b1b985f70f Mon Sep 17 00:00:00 2001
From: Cory Fields <theuni-nospam-@xbmc.org>
Date: Fri, 14 Sep 2012 15:35:32 -0400
Subject: [PATCH] d-r: enable dirty-regions by default

This enables the all-or-nothing mode, which means that we repaint the entire
screen if there's a single dirty region.

Also, nofliptimeout=0 is set to ensure that we never flip without rendering.

With these combined, this should be universally safe (and within spec) for all
gfx cards/drivers.

The main possible side-effect is the slow-motion effect brought about on slower
cpus that can't keep up at times and produce wild timestamps. With our improved
decoding and caching, this is now quite rare.
---
 xbmc/settings/AdvancedSettings.cpp |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/xbmc/settings/AdvancedSettings.cpp b/xbmc/settings/AdvancedSettings.cpp
index 47aaf7e..6f75227 100644
--- a/xbmc/settings/AdvancedSettings.cpp
+++ b/xbmc/settings/AdvancedSettings.cpp
@@ -306,8 +306,8 @@ void CAdvancedSettings::Initialize()
 
   m_canWindowed = true;
   m_guiVisualizeDirtyRegions = false;
-  m_guiAlgorithmDirtyRegions = 0;
-  m_guiDirtyRegionNoFlipTimeout = -1;
+  m_guiAlgorithmDirtyRegions = 3;
+  m_guiDirtyRegionNoFlipTimeout = 0;
   m_logEnableAirtunes = false;
   m_airTunesPort = 36666;
   m_airPlayPort = 36667;
-- 
1.7.10

